src/docbuild/cli/cmd_validate/process.py

```diff
@@ -168,6 +168,43 @@ async def validate_rng(
         )
 
 
+def validate_rng_lxml(
+    xmlfile: Path,
+    rng_schema_path: Path = PRODUCT_CONFIG_SCHEMA,
+) -> tuple[bool, str]:
+    """
+    Validate an XML file against a RELAX NG (.rng) schema using lxml.
+
+    This function uses lxml.etree.RelaxNG, which supports only the XML syntax
+    of RELAX NG schemas (i.e., .rng files, not .rnc files).
+
+    :param xmlfile: Path to the XML file to validate.
+    :param rng_schema_path: Path to the RELAX NG (.rng) schema file.
+    :return: Tuple of (is_valid: bool, error_message: str)
+    """
+    try:
+        # Parse the RELAX NG schema from .rng file
+        with open(rng_schema_path, 'rb') as f:
+            relaxng_doc = etree.parse(f)
+        relaxng = etree.RelaxNG(relaxng_doc)
+
+        # Parse the XML file to validate
+        with open(xmlfile, 'rb') as f:
+            xml_doc = etree.parse(f)
+
+        # Perform validation
+        is_valid = relaxng.validate(xml_doc)
+        if is_valid:
+            return True, ''
+        else:
+            # Return validation error log as string
+            return False, str(relaxng.error_log)
+
+    except Exception as e:
```

Comment:
Can you be more specific here? `Exception` catches a bit too much. IMHO you should catch these errors:

* XML syntax errors. This can happen either in the RNG or XML file.
* Validation errors. Happens when the XML file doesn't conform to the RNG schema.